src/quantuminspire/qiskit/backend_qx.py

```diff
@@ -173,17 +178,19 @@ def run(self,
         self.__validate_number_of_shots(number_of_shots)
 
         identifier = uuid.uuid1()
-        project_name = 'qi-sdk-project-{}'.format(identifier)
+        project_name = f'qi-sdk-project-{identifier}'
         project: Optional[Dict[str, Any]]
         project = self.__api.create_project(project_name, number_of_shots, self.__backend)
+        if not allow_fsp:  # when allow_fsp = False, we know what we do, no message needed
```

Comment:
I don't really understand what this comment means. "we know what we do" is very vague to me, and probably requires understanding what `show_fsp_warning` does exactly. Clearly if FSP is not allowed there is something to be aware of (maybe if you never specify a measure operation or something?). It would be good to clarify this comment.

Reply:
This flag is set to true only when the user added explicitly "allow_fsp=False" as a parameter to the Qiskit execute method to run the circuit. These non-qiskit parameters are stored in the run_config. In method run (in backend_qx) we handle allow_fsp when it was added to run_config.
In some cases (for example to compute quantum volume) the user never wants to run full state projection (an automatic optimization in our simulation backends) even when the circuit can be optimized. For that I added this flag. When it is false the user did that (that's why I added we know what we do....).
I added the following comment instead:
method run was called explicitly with allow_fsp = False. That is: full_state_projection is turned off.
No need to attend the user that the execution may take longer.